[Bug]: RDR test_failover_after_multiple_pods_failure fails with TimeoutExpiredError while waiting for VolumeSnapshot and leaves cluster unhealthy

### Description
RDR test_failover_after_multiple_pods_failure fails with TimeoutExpiredError while waiting for VolumeSnapshot and leaves cluster unhealthy that breaks the rest of the run

Test path: tests/functional/disaster-recovery/regional-dr/test_failover_after_multiple_pods_failure.py::TestFailoverAfterMultiplePodsFailure::test_failover_after_multiple_pods_failure

Error Details:
```
03:43:48 - MainThread - ocs_ci.utility.utils - ERROR - C[sagrawal-f18s] - function get_resource_count failed to return expected value 10 after multiple retries during 900 second timeout
03:43:48 - MainThread - ocs_ci.framework.pytest_customization.reports - INFO - C[sagrawal-f18s] - duration reported by tests/functional/disaster-recovery/regional-dr/test_failover_after_multiple_pods_failure.py::TestFailoverAfterMultiplePodsFailure::test_failover_after_multiple_pods_failure immediately after test execution: 3055.79
FAILED
_____________________________________________ TestFailoverAfterMultiplePodsFailure.test_failover_after_multiple_pods_failure ______________________________________________

self = <test_failover_after_multiple_pods_failure.TestFailoverAfterMultiplePodsFailure object at 0x14a725190>
dr_workload = <function create_workload_factory.<locals>.factory at 0x14a818180>, scale_deployments = <function scale_deployments.<locals>._scale at 0x14a819ee0>

    def test_failover_after_multiple_pods_failure(
        self,
        dr_workload,
        scale_deployments,
        # node_restart_teardown,
    ):
        """
        Tests to verify application failover between managed clusters when multiple deployments are scaled down on the
        primary managed cluster.

        """

        # Deploy Subscription and Appset based application of both RBD and CephFS SC
        rdr_workload = dr_workload(
            num_of_subscription=1,
            num_of_appset=1,
            pvc_interface=constants.CEPHBLOCKPOOL,
        )
        dr_workload(
            num_of_subscription=1,
            num_of_appset=1,
            pvc_interface=constants.CEPHFILESYSTEM,
        )
        drpc_objs = []
        for wl in rdr_workload:
            if wl.workload_type == constants.SUBSCRIPTION:
                drpc_objs.append(DRPC(namespace=wl.workload_namespace))
            else:
                drpc_objs.append(
                    DRPC(
                        namespace=constants.GITOPS_CLUSTER_NAMESPACE,
                        resource_name=f"{wl.appset_placement_name}-drpc",
                    )
                )

        primary_cluster_name = get_current_primary_cluster_name(
            rdr_workload[0].workload_namespace
        )
        secondary_cluster_name = get_current_secondary_cluster_name(
            rdr_workload[0].workload_namespace
        )

        # Verify the creation of ReplicationDestination resources on secondary cluster in case of CephFS
        config.switch_to_cluster_by_name(secondary_cluster_name)
        for wl in rdr_workload:
            if wl.pvc_interface == constants.CEPHFILESYSTEM:
                dr_helpers.wait_for_replication_destinations_creation(
                    wl.workload_pvc_count, wl.workload_namespace
                )
                # Verifying the existence of replication group destination and volume snapshots
                # in case of CG enabled for CephFS
                if dr_helpers.is_cg_cephfs_enabled():
                    dr_helpers.wait_for_resource_existence(
                        kind=constants.REPLICATION_GROUP_DESTINATION,
                        namespace=wl.workload_namespace,
                        should_exist=True,
                    )
                    dr_helpers.wait_for_resource_count(
                        kind=constants.VOLUMESNAPSHOT,
                        namespace=wl.workload_namespace,
                        expected_count=wl.workload_pvc_count,
                    )

        scheduling_interval = get_scheduling_interval(
            rdr_workload[0].workload_namespace, rdr_workload[0].workload_type
        )

        two_times_scheduling_interval = 2 * scheduling_interval  # Time in minutes
        time.sleep(two_times_scheduling_interval * 60)

        logger.info("Calling fixture to scale down deployments")
        config.switch_to_cluster_by_name(primary_cluster_name)
        scale_deployments("down")
        time.sleep(120)

        # Failover action via CLI
        logger.info("Failover workloads after pods failure")
        failover_results = []
        with ThreadPoolExecutor() as executor:
            for wl in rdr_workload:
                failover_results.append(
                    executor.submit(
                        failover,
                        failover_cluster=secondary_cluster_name,
                        namespace=wl.workload_namespace,
                        workload_type=wl.workload_type,
                        workload_placement_name=(
                            wl.appset_placement_name
                            if wl.workload_type != constants.SUBSCRIPTION
                            else None
                        ),
                    )
                )
                time.sleep(5)

        # Wait for failover results
        for fl in failover_results:
            fl.result()

        # Verify resources creation on secondary cluster (failoverCluster)
        config.switch_to_cluster_by_name(secondary_cluster_name)
        for wl in rdr_workload:
>           wait_for_all_resources_creation(
                wl.workload_pvc_count,
                wl.workload_pod_count,
                wl.workload_namespace,
            )

tests/functional/disaster-recovery/regional-dr/test_failover_after_multiple_pods_failure.py:142:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
ocs_ci/helpers/dr_helpers.py:1095: in wait_for_all_resources_creation
    wait_for_replication_resources_creation(
ocs_ci/helpers/dr_helpers.py:903: in wait_for_replication_resources_creation
    wait_for_resource_count(
ocs_ci/helpers/dr_helpers.py:808: in wait_for_resource_count
    sample.wait_for_func_value(expected_count)
ocs_ci/utility/utils.py:1843: in wait_for_func_value
    for i_value in self:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <ocs_ci.utility.utils.TimeoutSampler object at 0x14b04da50>

    def __iter__(self):
        if self.start_time is None:
            self.start_time = time.time()
        attempt = 0
        while True:
            self.last_sample_time = time.time()
            if self.timeout <= (self.last_sample_time - self.start_time):
>               raise self.timeout_exc_cls(*self.timeout_exc_args)
E               ocs_ci.ocs.exceptions.TimeoutExpiredError: Timed out after 900s running get_resource_count("kind"="volumesnapshot", "namespace"="sub-workload-d8e8ae7cc3e0-cephfs")

ocs_ci/utility/utils.py:1809: TimeoutExpiredError
```

--- a/tests/functional/disaster-recovery/regional-dr/conftest.py
+++ b/tests/functional/disaster-recovery/regional-dr/conftest.py
@@ -146,7 +146,11 @@ def scale_deployments(request):
         },
     ]
 
+    cluster_name = []
+
     def _scale(status="down"):
+        if status == "down":
+            cluster_name.append(config.current_cluster_name())
         replica_count = 0 if status == "down" else 1
         for dep in deployments_to_scale:
             try:
@@ -162,6 +166,9 @@ def scale_deployments(request):
 
     def teardown():
         log.info("Finalizer: scaling up deployments")
+        if cluster_name:
+            log.info(f"Switching to cluster '{cluster_name[0]}' before scaling up")
+            config.switch_to_cluster_by_name(cluster_name[0])
         _scale("up")
 
     request.addfinalizer(teardown)

--- a/tests/functional/disaster-recovery/regional-dr/test_failover_after_multiple_pods_failure.py
+++ b/tests/functional/disaster-recovery/regional-dr/test_failover_after_multiple_pods_failure.py
@@ -19,6 +19,7 @@ from ocs_ci.helpers.dr_helpers import (
     wait_for_all_resources_deletion,
 )
 from ocs_ci.ocs import constants
+from ocs_ci.ocs.node import get_nodes
 from ocs_ci.ocs.resources.drpc import DRPC
 from ocs_ci.ocs.resources.pod import wait_for_pods_to_be_running
 from ocs_ci.utility.utils import ceph_health_check
@@ -38,10 +39,7 @@ class TestFailoverAfterMultiplePodsFailure:
     """
 
     def test_failover_after_multiple_pods_failure(
-        self,
-        dr_workload,
-        scale_deployments,
-        node_restart_teardown,
+        self, dr_workload, scale_deployments, nodes_multicluster
     ):
         """
         Tests to verify application failover between managed clusters when multiple deployments are scaled down on the
@@ -143,10 +141,19 @@ class TestFailoverAfterMultiplePodsFailure:
                 wl.workload_pvc_count,
                 wl.workload_pod_count,
                 wl.workload_namespace,
+                performed_dr_action=True,
             )
 
-        logger.info("Calling fixture to scale up deployments")
         config.switch_to_cluster_by_name(primary_cluster_name)
+        logger.info(
+            "Restarting worker nodes on primary cluster to avoid pod sandbox creation failures "
+            "before scaling up deployments"
+        )
+        primary_cluster_index = config.cur_index
+        primary_worker_nodes = get_nodes()
+        nodes_multicluster[primary_cluster_index].restart_nodes(primary_worker_nodes)
+
+        logger.info("Calling fixture to scale up deployments")
         scale_deployments("up")
         wait_for_pods_to_be_running(
             namespace=constants.OPENSHIFT_STORAGE_NAMESPACE, timeout=420, sleep=30

